website/static/files/build_orca_reconcile.sql

```diff
@@ -6,150 +6,141 @@ SET search_path TO orca, public;
+  COMMENT ON COLUMN reconcile_jobs.error_message IS 'Critical error the job ran into that prevented it from finishing.';
 
-    COMMENT ON TABLE reconcile_s3_objects
-      IS 'Temporary table that holds the listing from the ORCA S3 bucket to use for comparisons against the ORCA catalog.';
-    COMMENT ON COLUMN reconcile_s3_objects.job_id
-      IS 'Job the S3 listing is a part of for the comparison. Foreign key to the reconcile jobs table.';
-    COMMENT ON COLUMN reconcile_s3_objects.orca_archive_location
-      IS 'ORCA S3 Glacier bucket name where the file is stored.';
-    COMMENT ON COLUMN reconcile_s3_objects.key_path
-      IS 'Full path and file name of the object in the S3 bucket from the S3 list command.';
-    COMMENT ON COLUMN reconcile_s3_objects.etag
-      IS 'AWS etag value from the S3 list command.';
-    COMMENT ON COLUMN reconcile_s3_objects.last_update
-      IS 'AWS Last Update from the S3 list command.';
-    COMMENT ON COLUMN reconcile_s3_objects.size_in_bytes
-      IS 'AWS size of the file in bytes from the S3 list command.';
-    COMMENT ON COLUMN reconcile_s3_objects.storage_class
-      IS 'AWS storage class the object is in from the S3 list command.';
 
+  CREATE TABLE reconcile_s3_objects  ( 
+    job_id               	int8 NOT NULL,
+    orca_archive_location	text NOT NULL,
+    key_path             	text NOT NULL,
+    etag                 	text NOT NULL,
+    last_update          	timestamp with time zone NOT NULL,
+    size_in_bytes        	int8 NOT NULL,
+    storage_class        	text NOT NULL,
+    delete_marker        	bool NOT NULL,
+    CONSTRAINT FK_reconcile_jobs_s3_objects FOREIGN KEY(job_id) REFERENCES reconcile_jobs(id)
+    ) PARTITION BY RANGE (orca_archive_location);
+  COMMENT ON TABLE reconcile_s3_objects IS 'Temporary table that holds the listing from the ORCA S3 bucket to use for comparisons against the ORCA catalog.
+  Partitions should be placed around orca_archive_location, followed by indexing on key_path once data is ingested.';
+  COMMENT ON COLUMN reconcile_s3_objects.job_id IS 'Job the S3 listing is a part of for the comparison. Foreign key to the reconcile jobs table.';
+  COMMENT ON COLUMN reconcile_s3_objects.orca_archive_location IS 'ORCA S3 Glacier bucket name where the file is stored.';
+  COMMENT ON COLUMN reconcile_s3_objects.key_path IS 'Full path and file name of the object in the S3 bucket from the S3 list command.';
+  COMMENT ON COLUMN reconcile_s3_objects.etag IS 'AWS etag value from the S3 list command.';
+  COMMENT ON COLUMN reconcile_s3_objects.last_update IS 'AWS Last Update from the S3 list command.';
+  COMMENT ON COLUMN reconcile_s3_objects.size_in_bytes IS 'AWS size of the file in bytes from the S3 list command.';
+  COMMENT ON COLUMN reconcile_s3_objects.storage_class IS 'AWS storage class the object is in from the S3 list command.';
+  COMMENT ON COLUMN reconcile_s3_objects.delete_marker IS 'Set to `True` if object is a delete marker.';
```

Comment:
You will need to add the partitions. I would dummy some up with comments for this SQL, in the db_deploy, we will need to make it dynamic based on the orca buckets they actually have.

```
create table reconcile_s3_objects_<s3 Bucket name goes here>
partition of reconcile_s3_objects
for values in ('<s3 Bucket name goes here>');
 
COMMENT ON TABLE reconcile_s3_objects_<s3 Bucket name goes here> IS 'reconcile_s3_objects table partition with orca_archive_location value of <s3 Bucket name goes here>.'

.... Other Partitions ....

create table reconcile_s3_objects_default
    partition of reconcile_s3_objects DEFAULT;

COMMENT ON TABLE reconcile_s3_objects_default IS 'reconcile_s3_objects table partition that captures any orca_archive_location values without a partition.'
```

Reply:
We shouldn't ever use a default table.
Apparently I am unclear on the flow. If we drop a partition, don't we need to recreate it?

Reply:
The default is a catch all so that no records fall through the cracks. Ideally, nothing would go to default but it is a good item to have to let us know if something is not correct with the partitions or settings and provides an opportunity for use to easily recover data. Without the default, you will get an insert error.

Yes if you drop a partition you need to recreate  it, but you need to create the partition to begin with.

Reply:
I would prefer a loud error, rather than a silent one that we have to check the db to find.

Since we have to create the partition each time we pull in data, we should do that there rather than creating partitions we will never use in db_deploy.

Reply:
Switching to new plans to use Truncate. Adding example SQL to db_deploy card.